playbooks/common/openshift-cluster/upgrades/v3_3/upgrade.yml

```diff
@@ -39,8 +39,9 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when:
+      - openshift_http_proxy is defined or openshift_https_proxy is defined
+      - openshift_generate_no_proxy_hosts | default(True) | bool
```

Comment:
wish there were an easy way to apply this same logic generally instead of copying it around... maybe a filter. anyway that's not the point of this PR.